### PR TITLE
New/currency rate editor

### PR DESCRIPTION
**Pull request by atm-jpb**
# New multi-currency exchange rate editor
## Context
Currently multicurrency logs past exchange rates when you "update" the exchange rate of a currency, but it doesn't enable you to correct past exchange rates or to date an exchange rate when you add it (the date is automatically set to the current time).

## This PR
This PR introduces a new page that enables you to view, edit or add exchange rates in a way somewhat similar to Dolibarr dictionaries. The page is accessible via the administration module.

image

## questions 
    you'll  see in multicurrency_rate.php  on  line 205 
 the action_massactions.inc.php need $uploaddir to operate but not really needed in multicurrency context. my question is this one : do i have to create this directory  ?

### CODE REVIEW

**Comment by eldy** [comment]
No, if there is no massaction implemented, no need to create the directory. Just set a value.

**Comment by atm-jpb** [comment]
@eldy that's what i did : $uploaddir = $conf->multicurrency->multidir_output; // define only because core/actions_massactions.inc.php want it
it was just a question , then could you review/merge my pr ?

**Comment by eldy** [comment]
The PR has conflicts making merge not possible. Can you fix them ?

File: htdocs/langs/fr_FR/multicurrency.lang [hunk 1]
```diff
@@ -18,3 +18,21 @@ MulticurrencyReceived=Reçu, devise originale
 MulticurrencyRemainderToTake=Montant restant, devise d'origine
 MulticurrencyPaymentAmount=Montant du règlement (devise d'origine)
 AmountToOthercurrency=Montant destination (en devise du compte de réception)
+TabTitleMulticurrencyRate = liste des Taux
```
Comment (frederic34) [inline htdocs/langs/fr_FR/multicurrency.lang:21]: new lang are created in en_US, other lang are overwrited by transifex

**Comment by atm-jpb** [comment]
i fixed the conflict and create a new pr
https://github.com/Dolibarr/dolibarr/pull/15116

**Pull request closed without merging**